soc/st/stm32/stm32wbax/power.c

```diff
@@ -115,10 +115,6 @@ static void set_mode_suspend_to_ram(void)
 
 	/* Save context and enter Standby mode */
 	arch_pm_s2ram_suspend(suspend_to_ram);
-
-	/* Execution is restored at this point after wake up */
-	/* Restore system clock as soon as we exit standby mode */
-	sys_clock_idle_exit();
```

Comment:
This call was made here to restore the system clock as soon as possible when exiting STM32 Standby mode (by calling sys_clock_driver_init first in sys_clock_idle_exit). When in this state, the LPTIM is shut off and if we don't reenable it, we don't have a system clock, and all functions like sleep or similar don't work properly.

I've tested the sample boards/stm32/power_mgmt/suspend_to_ram/ on a nucleo_wba55cg and it doesn't work anymore.

I understand if the function sys_clock_idle_exit is not be the best place to re-init our system clock.
What I'm sure is that we have to do it here, just after wakeup. Do you have any idea?

Reply:
If the default call to sys_clock_idle_exit() is too late for your hardware, the correct trick would presumably be to have a driver-internal API exposed (e.g. check CONFIG_STM32_whatever to see that it's your particular timer driver and just call directly into it), then stub the driver's implementation of sys_clock_idle_exit() in the context where it knows it will be called out of the SOC resume code and not the PM layer.

Reply:
Does it hurt to keep it here ?
As we're already dealing with some conditions in this function implementation of the driver:
https://github.com/zephyrproject-rtos/zephyr/blob/ced9fba95c9d68e7354bd9c589ac81844784abd8/drivers/timer/stm32_lptim_timer.c#L553-L563

Reply:
It's his patch and I'll let @ceolin make the call, but generally subsystem cleanup is done precisely because weirdness like this "hurts" a bit.  APIs and frameworks lose their value when the rules aren't followed.  And in this case it looks like the suggestion would work fine: make sys_clock_idle_exit() a stub (actually there might already be a __weak default?), rename the code body to "stm32_lptim_standby_exit()" or whatever, and call that out of the SOC layer.

Reply:
@andyross I really liked the suggestion. Thanks !